### ISSUE TITLE
Firefox account registration dead flow on Fenix

### ISSUE THREAD

**Issue by vasilicatamas** [mozilla/fxa#4714]
**Build:**
Prod [Train 1.164.0]

**Affected platforms:**
Android 9

**Steps to reproduce:**
1. Install Firefox and launch the app.
2. Go to menu -> Settings -> Turn on Sync
3. Tap on "Use email instead" button.
4. Insert valid credentials for email address, password and age fields.
5. Taps on "Create account" button.
6. Choose any of the options and tap on "Sync selections" button.

**Expected results:**
The user is able to navigate to another page in order to confirm the account.

**Actual Results:**
The toolbar is not displayed and there is no way to leave the "Enter verification code" page.

![deadflow](https://user-images.githubusercontent.com/18114210/78149801-14443f00-743f-11ea-9f79-790d1fb9f7ab.gif)

**Comment by vladikoff** [comment]
> Actual Results:
The toolbar is not displayed and there is no way to leave the "Enter verification code" page.

@vasilicatamas sorry could you explain what do you mean by that?

your gif seems to end at the verification code page? are you not able to type in the verification code?

**Comment by vasilicatamas** [comment]
I need to switch to another tab in order to copy the verification code and I am unable to leave the "Enter verification code" page because the url bar is not displayed.

Here's what happens for login flow, but even here the user must click on those 3 dots, select open in Firefox Nightly, go back through all "Turn on Sync" Fenix flow until the user gets to the main tabs page in order to be able to open a new tab to navigate to the email inbox.

![Screenshot_23](https://user-images.githubusercontent.com/18114210/78153646-f4fbe080-7443-11ea-8d79-f420ff01fa0d.png)

**Comment by vladikoff** [comment]
> there is no way to leave

@vasilicatamas oh I see. How about the Android "Back" button?

**Comment by vladikoff** [comment]
> I need to switch to another tab in order to copy the verification code and I am unable to leave the "Enter verification code" page because the url bar is not displayed.

Oh sorry this comment didn't load. I see what you mean about opening another tab though

**Comment by vladikoff** [comment]
I wonder if it makes more sense to transfer this issue to https://github.com/mozilla-mobile/fenix I don't think the FxA front-end server can do anything about fixing this flow

**Comment by vladikoff** [comment]
cc @grigoryk

**Comment by grigoryk** [comment]
https://github.com/mozilla-mobile/fenix is a better place for this.

However, this is functioning "as designed". We use a "custom tab" for the sign-in flow, which takes over the browser UX and acts as a modal dialog. The expectation is that user will either use their phone's email client, or another computer/browser altogether (e.g. firefox on their laptop).
I see how sometimes this is non-ideal. The workaround here would be to pre-installed browser on the phone, which seems fine to me.

**Comment by vladikoff** [comment]
> The workaround here would be to pre-installed browser on the phone, which seems fine to me.

Yeah that's what i am thinking too, I don't think that many people use "web-based" email on their phone. Therefore we can probably change the severity of this bug.

**Comment by data-sync-user** [comment]
➤ John Gruen commented:

Removed the fpn mobile epic since that's just about subscription UI

**Comment by data-sync-user** [comment]
➤ Wil Clouser commented:

[~pwalmsley@mozilla.com] - what should we do here?  /cc [~adavis@mozilla.com]

**Comment by data-sync-user** [comment]
➤ Alex Davis commented:

I'm not 100% sure that this is a bug or a problem. Most phones (if not all) have email setup in a mail client.

I just read Grisha's comment after writing my own and I agree with him. This feels like an edge case that we can maybe dig into with surveys in the future.
